### PR TITLE
fix(route): AP News获取方式调整

### LINKED ISSUE #12733: APNews "Cannot read properties of null (reading '1')"
### Routes

```routes
/apnews/topics/:topic?
```

### Full routes

```fullroutes
/apnews/topics/:topic?
```

### Related documentation

https://docs.rsshub.app/en/traditional-media.html#ap-news

### What is expected?

RSS feed XML data is returned

### What is actually happening?

Route requested: /topics/apf-topnews
Error message: Response code 404 (Not Found): target website might be blocking our access, you can [host your own RSSHub instance](https://docs.rsshub.app/install/) for a better usability.
Helpful Information to provide when opening issue:
Path: /topics/apf-topnews
Node version: v18.16.1
Git Hash: 7775ee5

### Deployment information

RSSHub demo (https://rsshub.app)

### Deployment information (for self-hosted)

_No response_

### Additional info

```shell
N/A
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by dzx-dzx**
## 该 PR 相关 Issue / Involved Issue

Close #12733 

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/apnews/topics/abortion
/apnews/topics/china
/apnews/topics/biology
/apnews/topics/trending-news
/apnews/topics/ap-top-news
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Please remove the unused code.

File: lib/v2/apnews/topics.js [hunk 1]
```diff
@@ -1,45 +1,37 @@
...
+        }))
+        .map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await got(item.link);
+                const $ = cheerio.load(response);
+
+                return Object.assign(item, {
+                    pubDate: new Date($("meta[property='article:published_time']").attr("content")),
+                    updated: new Date($("meta[property='article:modified_time']").attr("content")),
+                    description: $('div.RichTextStoryBody').html(),
```
Comment (TonyRL) [inline lib/v2/apnews/topics.js:25]: Please remove the "Other news"(`div.Enhancement`) from `description`

Comment (dzx-dzx) [reply lib/v2/apnews/topics.js:25]: I've thought about this, but is there any elegant way...?

Comment (TonyRL) [reply lib/v2/apnews/topics.js:25]: `$('div.Enhancement').remove();` before returning.

Comment (dzx-dzx) [reply lib/v2/apnews/topics.js]: https://github.com/DIYgod/RSSHub/blob/bb558b3502ee29bfb13508cedbb41a9c0accb309/lib/v2/apnews/topics.js#L26

Do you think there's a better way? I didn't manage to remove the element using `filter()` or `not()`.

File: lib/v2/apnews/topics.js [hunk 2]
```diff
@@ -1,45 +1,39 @@
...
+            }))
+            .map((item) =>
+                ctx.cache.tryGet(item.link, async () => {
+                    const { data: response } = await got(item.link);
+                    const $ = cheerio.load(response);
+
+                    return Object.assign(item, {
+                        pubDate: new Date($("meta[property='article:published_time']").attr('content')),
+                        updated: new Date($("meta[property='article:modified_time']").attr('content')),
+                        description: ($('div.RichTextStoryBody').find('.Enhancement').remove().html(), $('div.RichTextStoryBody').html()),
```
Comment (TonyRL) [inline lib/v2/apnews/topics.js:26]: `$('div.Enhancement').remove();` **before** returning.

_Originally posted by TonyRL in https://github.com/DIYgod/RSSHub/pull/12812#discussion_r1264692853_
            
```suggestion
                    $('div.Enhancement').remove();
                    return Object.assign(item, {
                        pubDate: new Date($("meta[property='article:published_time']").attr('content')),
                        updated: new Date($("meta[property='article:modified_time']").attr('content')),
                        description: $('div.RichTextStoryBody').html(),
```

Comment (dzx-dzx) [reply lib/v2/apnews/topics.js]: Well I personally prefer oneliner code... Sorry.

**Pull request merged**